packages/shared/lib/services/connections/credentials/refresh.ts

```diff
@@ -182,6 +186,7 @@ async function refreshCredentials(
                 connection: { id: oldConnection.id, name: oldConnection.connection_id }
             }
         );
+        void flushLogsBuffer((logsBuffer.transport as BufferTransport).buffer, logCtx);
```

Comment:
should flushing be abstracted and happen automatically on` logCtx.failed/success` if transport is Buffer. With maybe a `forceFlush` method in `LogContextStateless` to manually flush buffer context if needed

Reply:
we would need to attach/pass the buffer to the actual logCtx that contains the proper transport.
Maybe we could have a `logCtx.merge(logCtx)` and the transport decide on what to do: 
- if buffer, it saves more buffer
- if es, it flushes everything
Wdyt?

Reply:
I am not sure why we create two logCtx in this case. Can we not reuse the buffer one?

Reply:
rn the buffer doesn't have the capability to create an operation, just for simplicity's sake, it's just buffering the logs but we could improve for sure

Reply:
I see. a merge `logCtx.merge(logsBuffer)` could be a good idea indeed. this or passing a buffer to `logContextGetter.create`

Reply:
Changed in [450e17a](https://github.com/NangoHQ/nango/pull/3765/commits/450e17a7c5a539bf0937cd5c377c214e3a93f359)